pkg/cluster/delete.go

```diff
@@ -356,6 +357,27 @@ func (m *manager) deleteGateway(ctx context.Context) error {
 	return nil
 }
 
+func (m *manager) deleteClusterMsiCertificate(ctx context.Context) error {
+	// The cluster MSI may have been deleted prior to cluster deletion. If that's the case
+	// we will have already deleted the certificate.
+	if m.doc.OpenShiftCluster.Identity == nil || m.doc.OpenShiftCluster.Identity.UserAssignedIdentities == nil || len(m.doc.OpenShiftCluster.Identity.UserAssignedIdentities) == 0 {
+		m.log.Warning("skipping cluster MSI certificate deletion because cluster MSI has already been deleted")
+		return nil
+	}
+
+	secretName, err := m.clusterMsiSecretName()
+	if err != nil {
+		return err
+	}
+
+	err = m.clusterMsiKeyVaultStore.DeleteCredentialsObject(ctx, secretName)
+	if azcoreErr, ok := err.(*azcore.ResponseError); !ok || azcoreErr.StatusCode != 404 {
```

Comment:
Should we check if `err != nil` here first?

Reply:
I did thorough unit test cases here, and it works correctly when the error is non-nil and not an `azcore.ResponseError`.

But now that you've asked this question, I see that even though it works correctly it reads unintuitively. I'll think of a way to simplify it so that it reads better.

Reply:
I pushed an update here. Let me know if this didn't help!